### PR TITLE
Enforce "good" states of binary_sensor as green in history timeline chart

### LINKED ISSUE #8144: Raspberry Pi Power Supply Checker - Widget Status Colors Switched
## The problem

I think the status of the "Raspberry Pi Power Supply Checker" integration shows inverse colors (green if there is a problem and red if the status says OK). More specifically it's the "RPi Power status" that shows an incorrect color coding on lovelace.

Screenshot: [Link](https://paste.pics/B9Y0O)

## Environment

- Home Assistant Core release with the issue: 5.10
- Last working Home Assistant Core release (if known): unknown, integration was not used previously
- Operating environment (OS/Container/Supervised/Core): hassio latest stable
- Integration causing this issue: Raspberry Pi Power Supply Checker
- Link to integration documentation on our website: [](https://www.home-assistant.io/integrations/rpi_power)

## Problem-relevant `configuration.yaml`

```yaml

```

## Traceback/Error logs

```txt

```

### PR DESCRIPTION
**Pull request by spacegaier**
## Proposed change

I implemented a logic where we can define the device_classes for binary_sensor that should use the inverted color (where "off" = "good"). For a few (e.g. motion) is cannot really be defined, so I left them as is and only added the ones where it is pretty clear.

In an ideal world the user could define that per entity what is good and what is bad.

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes https://github.com/home-assistant/frontend/issues/8144
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by bramkragten — APPROVED** [review]

**Comment by bramkragten** [comment]
Shouldn't that be a binary sensor?

**Comment by spacegaier** [comment]
> 
> 
> Shouldn't that be a binary sensor?

You lead me to realize a good point: Yes it is a binary sensor with device class "problem". Here "off" means "OK" (should be green), whereas "on" means "Problem". So we would need to invert the colors for this and some others (e.g. "safety").

**Comment by spacegaier** [comment]
@bramkragten I now implemented a logic where we can define the device_classes for binary_sensor that should use the inverted color (where "off" = "good"). For a few (e.g. motion) is cannot really be defined, so I left them as is and only added the ones where it is pretty clear.

In an ideal world the user could define that per entity what is good and what is bad.

File: src/common/const.ts [hunk 1]
```diff
@@ -127,6 +127,24 @@ export const STATES_OFF = ["closed", "locked", "off"];
 export const BINARY_STATE_ON = "on";
 export const BINARY_STATE_OFF = "off";
 
+/** Binary sensor device classes for which the static colors for on/off need to be inverted.
+ *  List the ones were "off" = good or normal state = should be rendered "green".
+ */
+export const BINARY_SENSOR_DEVICE_CLASS_COLOR_INVERTED = new Set([
```
Comment (bramkragten) [inline src/common/const.ts:133]: Since this is specific to the carts, it should live there

**Pull request merged**